### PR TITLE
HPCC-22686 Update supercopy.ecl to test copy for super index with <=1 and >1 subfile

### PR DESCRIPTION
**Pull request by AttilaVamos**
Add tests for:

   - Copy superfile with 0, 1, 2 subfiles as logical and as superfile
   - Copy superindex with 0, 1, 2, 3 subfiles as logical and as superfile
   - Check the target file is the same as expected or as it forced to be (for superindex >=1 subfile)

Fix a DFUServer bug related to copy a superfile without subfile.

Signed-off-by: Attila Vamos <attila.vamos@gmail.com>

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [x] I have added tests to cover my changes.
  - [x] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [x] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [x] Send notifications about my Pull Request position in Smoketest queue.
- [x] Test my draft Pull Request.

## Testing:

Tested manually

### CODE REVIEW

**Comment by AttilaVamos** [comment]
@jakesmith please review.

**Comment by jakesmith** [comment]
@AttilaVamos comment says "I expect it will fail until HPCC-22670 Pr-12891 has been merged.", but it has been merged..

**Comment by jakesmith** [comment]
@AttilaVamos - also tick boxes need filling

**Comment by AttilaVamos** [comment]
@jakesmith Yes, HPCC-22670 Pr-12891 has been merged after I created this PR. 
Boxes ticked.

**Comment by jakesmith** [comment]
@AttilaVamos - So can you removed the now redundant comment form the commit and PR message (and also it's mentioned in a comment in the regression query)

File: testing/regress/ecl/supercopy.ecl [hunk 1]
```diff
@@ -88,116 +145,270 @@ rec supercopy(rec l) := TRANSFORM
...
+                                 SELF.result := dst[1].result + ' Fail',
                                  SELF.msg := FAILMESSAGE,
-                                 SELF.sourceFileName := superFileName,
-                                 SELF.destFileName := superFileCopyName,
-                                 SELF.destCluster := 'mythor',
-                                 SELF.asSuperfile := False
+                                 SELF.sourceFileName := dst[1].sourceFileName,
+                                 SELF.destFileName := dst[1].destFileName,
+                                 SELF.destCluster := dst[1].destCluster,
+                                 SELF.asSuperfile := dst[1].asSuperfile
```
Comment (jakesmith) [inline testing/regress/ecl/supercopy.ecl]: It's a bit odd to be passing in a dataset (with must have 1 row), when really regular parameters are wanted.
I think you could avoid by making both doSuperCopy and supercopy functions that take string/boolean parameters.
Then the callers wouldn't need to create a pseudo dataset to pass parameters in.

**Comment by jakesmith** [comment]
@AttilaVamos - looks ok, but It I think supercopy.ecl using dataset as parameter could be simplified, see example changes in IM.

**Comment by AttilaVamos** [comment]
Reopened in https://github.com/hpcc-systems/HPCC-Platform/pull/12926

**Pull request closed without merging**